### ISSUE TITLE
VueX stores data as Object, not as the Typescript Class

### ISSUE THREAD

**Issue by MichaelTamaki** [calband/calchart-redesign#36]
Issue: We are unable to call class methods on `show` in the store.

Possible solution: refactor `models` to be `interface` rather than `class`. Then, move the class methods into their own functions (in `mutations`?). Will have to investigate.

**Comment by MichaelTamaki** [comment]
https://stackoverflow.com/questions/54345327/is-using-es6-classes-in-vue-vuex-flux-an-anti-pattern

Seems like we should be thinking of VueX store as JSON, e.g. just data. This probably means that we can re-organize `models` to be like the following:
- `src/models`: Just `interface`s that describe the data.
- `src/logic`: For models that need it, introduce "class methods" here.

**Comment by rmpowell77** [comment]
I'm not sure I would do that.  I would keep these things together.  The properties of a class are the interface.  I always think that when you declare a property like `title`, you are effectively saying there are "logic" functions `getTitle` and `setTitle`, along with a data member called `title` that needs to be saved.

I think we're going to find as we add more editing functionality that there's going to be a lot of things that aren't easily modeled by properties.  Almost certainly there's going to be a lot of logic interface we're going to add on top to make sure that invariants are preserved; for example, setting a point's position may have to have a check to make sure the number isn't some insane value, like 1000000, because if that happens, the point will vanish off the screen and the user won't be able to move it back into position.  So having a clipper or some sort of bounding box function that overrides the `setPosition` property function is going to be needed.

I would keep the concept of having `src/models` in the project, but just accepting that there are some values that the model has that are modeled great with pure property type interface, and then there are some which need custom functions.

**Comment by MichaelTamaki** [comment]
Discussed on Slack: here's the proposed solution. We'll investigate alternatives again once I have a PR in.

Old Code:
```js
class ExampleClass {
  title: string  constructor(title: string) {
    this.title = title
  }  getTitle(): string {
    return this.title
  }  setTitle(title: string) {
    if (title === null) throw 'its null'
    this.title = title
  }
}
```

New Code: 
```js
interface ExampleClass {
  title: string
}

const ExampleClassLogic = {
  create(title: string): ExampleClass {
    return {
      title
    }
  }
  getTitle(example: ExampleClass): string {
    return example.title
  }
  setTitle(example: ExampleClass, title: string) {
    if (title === null) throw 'its null'
    example.title = title
  }
}
```

**Comment by MichaelTamaki** [comment]
https://github.com/vuejs/vuex/issues/153

**Comment by rmpowell77** [comment]
was this fixed in #37 ?
